packages/react-next/src/components/TeachingBubble/TeachingBubbleContent.base.tsx

```diff
@@ -86,6 +64,26 @@ export class TeachingBubbleContentBase extends React.Component<ITeachingBubblePr
       secondaryButtonClassName: secondaryButtonProps ? secondaryButtonProps.className : undefined,
     });
 
+    const onKeyDown = React.useCallback(
+      (ev: React.KeyboardEvent<HTMLElement> | KeyboardEvent): void => {
+        if (props.onDismiss) {
+          // eslint-disable-next-line deprecation/deprecation
+          if (ev.which === KeyCodes.escape) {
+            props.onDismiss(ev);
+          }
+        }
+      },
+      [props.onDismiss],
+    );
+
+    useOnEvent(getDocument(rootElementRef.current), 'keydown', onKeyDown as (ev: Event) => void);
```

Comment:
I'm not sure this will work--`rootElementRef.current` won't be defined yet on initial render, so finding the relevant document won't work correctly. Instead I think this will need to be wrapped in `useEffect`.